core/message/message.go

```diff
@@ -1,32 +1,23 @@
-package eventually
+package message
 
-// Payload is a Message payload.
+// Message is a Message payload.
 //
 // Each payload should have a unique name identifier, that can be used
 // to uniquely route a message to its type.
-type Payload interface {
+type Message interface {
 	Name() string
 }
 
-// Message represents any kind of information that can be carried around.
-//
-// Usually, a Message only contains a payload, but it could optionally
-// include some metadata (e.g. some debug identifiers).
-type Message struct {
-	Payload  Payload
-	Metadata Metadata
-}
-
 // Metadata contains some data related to a Message that are not functional
 // for the Message itself, but instead functioning as supporting information
 // to provide additional context.
-type Metadata map[string]interface{}
+type Metadata map[string]string
```

Comment:
Yeah, I think it makes more sense to provide a well-defined type here and accept the "limitations", rather than introducing a potential source for troubles with `interface{}` (or `any`) 👀